dockstore-integration-testing/src/test/java/io/dockstore/webservice/metrics/MetricsDataS3ClientIT.java

```diff
@@ -56,137 +74,181 @@
+
+        String id = "#workflow/github.com/DockstoreTestUser2/dockstore_workflow_cnv";
+        String versionId = "master";
+        String platform = Partner.TERRA.name();
+        String description = "A single execution";
+
+        // Test malformed ID
+        ApiException exception = assertThrows(ApiException.class, () -> extendedGa4GhApi.executionMetricsPost(List.of(), platform, "malformedId", "malformedVersionId", null));
+        assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), exception.getCode());
+
+        // Test ID that doesn't exist
+        exception = assertThrows(ApiException.class, () -> extendedGa4GhApi.executionMetricsPost(List.of(), platform, "github.com/nonexistent/id", "master", null));
+        assertEquals(Response.Status.NOT_FOUND.getStatusCode(), exception.getCode(), "Should not be able to submit metrics for non-existent id");
+        assertTrue(exception.getMessage().contains("Tool not found"), "Should not be able to submit metrics for non-existent id");
+
+        // Test version ID that doesn't exist
+        Workflow workflow = workflowApi.manualRegister("github", "DockstoreTestUser2/dockstore_workflow_cnv", "/workflow/cnv.cwl", "", "cwl",
+                "/test.json");
+        workflow = workflowApi.refresh1(workflow.getId(), false);
+        workflowApi.publish1(workflow.getId(), CommonTestUtilities.createOpenAPIPublishRequest(true));
+        exception = assertThrows(ApiException.class, () -> extendedGa4GhApi.executionMetricsPost(List.of(), platform, id, "nonexistentVersionId", null));
+        assertEquals(Response.Status.NOT_FOUND.getStatusCode(), exception.getCode(), "Should not be able to submit metrics for non-existent version");
+        assertTrue(exception.getMessage().contains("Version not found"), "Should not be able to submit metrics for non-existent version");
+
+        // Test that a non-admin/non-curator user can't submit metrics
+        exception = assertThrows(ApiException.class, () -> otherExtendedGa4GhApi.executionMetricsPost(createExecutions(1), platform, id, versionId, description));
+        assertEquals(Response.Status.FORBIDDEN.getStatusCode(), exception.getCode(), "Non-admin and non-curator user should not be able to submit metrics");
+
+        // Test that the response body must contain ExecutionStatus
+        List<Execution> executions = createExecutions(1);
+        executions.forEach(execution -> execution.setExecutionStatus(null));
+        exception = assertThrows(ApiException.class, () -> extendedGa4GhApi.executionMetricsPost(executions, platform, id, versionId, description));
+        assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), exception.getCode(), "Should not be able to submit metrics if ExecutionStatus is missing");
+        assertTrue(exception.getMessage().contains("must contain ExecutionStatus"), "Should not be able to submit metrics if ExecutionStatus is missing");
```

Comment:
ditto, basically the idea is to avoid a "refactor -> run tests and discover failure -> fix and re-submit" cycle, by making it more obvious what test by renaming/tweaking of the String error message, we can make it more obvious or even simply avoid the cycle